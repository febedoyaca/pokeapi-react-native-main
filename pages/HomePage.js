import React, {useState,useRef} from 'react';
import {StyleSheet, View, ImageBackground, Image, Button, Dimensions} from 'react-native';


export default function HomePage({navigation}) {
    const pokemonHandler = () =>{navigation.navigate('All Pokemon')}
    const searchHandler = () =>{navigation.navigate('Search Pokemon')}


    return(
        <View style={styles.container}>
            <ImageBackground source={{uri: 'https://cdn.wallpapersafari.com/99/25/rQeFHW.jpg'}} style={[styles.fixed, {zIndex:-1}]}>
            <View style={styles.button}>
                <Image source={{uri:'https://raw.githubusercontent.com/PokeAPI/media/master/logo/pokeapi_256.png'}} style={[styles.imageFront, {zIndex:0}]}/>
                <View style={{marginBottom:20}}></View>
                <Button title='All pokemon' onPress={pokemonHandler} color='black' />
                <View style={{marginBottom:20}}></View>
                <Button title='Search Pokemon' onPress={searchHandler} color='black' />
            </View>
            </ImageBackground>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignContent: 'center',
        justifyContent: 'center',
        alignItems:'center',
    },
    fixed: {
        position:'absolute',
        top:0,
        left:0,
        right:0,
        bottom:0,
        flex:1,
        width: Dimensions.get('window').width,
        height: Dimensions.get('window').height,
    },
    button: {
        width:200,
        flex:1,
        justifyContent:'center',
        alignSelf:'center',
        marginBottom:170,
    },
    imageFront:{
        width:257,
        height:103,
        justifyContent:'center',
        alignSelf:'center',
    }
})