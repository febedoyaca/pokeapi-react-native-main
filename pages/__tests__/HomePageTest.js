import React from "react";
import HomePage from '../HomePage';
import renderer from 'react-test-renderer';

it ('renders correctly', () => {
    const tree = renderer.create(<HomePage/>);
    expect(tree).toMatchSnapshot();
});