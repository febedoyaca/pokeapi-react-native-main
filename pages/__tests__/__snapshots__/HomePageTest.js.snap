// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Array [
        Object {
          "bottom": 0,
          "flex": 1,
          "height": 1334,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        },
        Object {
          "zIndex": -1,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://cdn.wallpapersafari.com/99/25/rQeFHW.jpg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignSelf": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginBottom": 170,
          "width": 200,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://raw.githubusercontent.com/PokeAPI/media/master/logo/pokeapi_256.png",
          }
        }
        style={
          Array [
            Object {
              "alignSelf": "center",
              "height": 103,
              "justifyContent": "center",
              "width": 257,
            },
            Object {
              "zIndex": 0,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "marginBottom": 20,
          }
        }
      />
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "black",
                },
              ]
            }
          >
            All pokemon
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 20,
          }
        }
      />
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "black",
                },
              ]
            }
          >
            Search Pokemon
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
